### ISSUE TITLE
Can't use expo publish url with custom dev client

### ISSUE THREAD

**Issue by hannojg** [expo/expo#13711]
### Summary

Generating a bundle with `expo publish` gives me an URL that has the format `https://exp.host/@hannojg/.../index.exp?sdkVersion=42.0.0`. Entering this URL in the custom development client opens the error screen instead of the app.

### Managed or bare workflow? If you have made manual changes inside of the `ios/` or `android/` directories in your project, the answer is bare!

bare, not using eas but building the app ourselves

### What platform(s) does this occur on?

Android, iOS

### Package versions

"expo": "~42.0.1",
"expo-dev-client": "^0.4.5",
"expo-updates": "~0.8.1",

### Environment

  Expo CLI 4.8.1 environment info:
    System:
      OS: macOS 11.4
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 15.14.0 - ~/.nvm/versions/node/v15.14.0/bin/node
      Yarn: 1.22.4 - /usr/local/bin/yarn
      npm: 7.7.6 - ~/.nvm/versions/node/v15.14.0/bin/npm
      Watchman: 2021.06.07.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.10.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
      Android SDK:
        API Levels: 23, 25, 26, 27, 28, 29, 30
        Build Tools: 27.0.3, 28.0.3, 29.0.2, 29.0.3, 30.0.0, 30.0.0, 30.0.1, 30.0.2, 30.0.3
        System Images: android-23 | Intel x86 Atom, android-30 | Google APIs Intel x86 Atom, android-R | Google APIs Intel x86 Atom
        Android NDK: 20.1.5948944
    IDEs:
      Android Studio: 4.2 AI-202.7660.26.42.7322048
      Xcode: 12.5.1/12E507 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~42.0.1 => 42.0.1 
      react: 16.13.1 => 16.13.1 
      react-dom: 16.13.1 => 16.13.1 
      react-native: ~0.63.4 => 0.63.4 
      react-native-web: ~0.13.12 => 0.13.18 
    npmGlobalPackages:
      expo-cli: 4.8.1
    Expo Workflow: bare

### Reproducible demo or steps to reproduce from a blank project

* Just set up a project with a custom dev client according to the docs https://docs.expo.io/clients/getting-started/
* Run `expo publish`
* Try to use the published URL in the custom dev client

** Detailed reproduction: **

```bash
expo init (we use typescript, blank)
yarn add expo-dev-client
expo run:ios  # creates ios folder
expo run:android # creates android folder

# Opening the iOS project with xcode and running the app on a physical device

expo publish

# taking the URL and inserting it to the development client
```

Interestingly you can use the URL of the bundle directly (extracted from the https://exp.host/...).
<div>
<img src="https://user-images.githubusercontent.com/16821682/126444957-ce8ec9d3-16cd-40f3-afad-6d12ba611d2e.jpeg" width=300 style="float: left;" />
<img src="https://user-images.githubusercontent.com/16821682/126444953-d28f6048-f5fe-49ee-bad9-de3ddc084c46.jpeg" width=300 style="float: right;" />
</div>

### Stacktrace (if a crash is involved)

* On android the custom dev client crashes and restarts
* On iOS we get the error which you can see in the attached files

**Comment by hannojg** [comment]
By the way, we are not using eas. We are building the native apps ourselves. Thus we are also looking to share the app bundle without eas.

**Comment by lukmccall** [comment]
Hi @hannojg, I was able to reproduce that behavior. We're curently working on this issue.

**Comment by hannojg** [comment]
I am not sure whether this is related to this issue, but `expo publish` also seems to just create a bundle for iOS, not for android (when inspecting the json, there is just a ios URL). If it's a different issue, I will open a new one.

**Comment by esamelson** [comment]
Hi @hannojg - we should have a fix for this issue soon. As a workaround, you should already be able to open your published project using the link that's displayed in the terminal by `expo publish` (just avoid the one that's actually hyperlinked to, with the extra `/index.exp....`).

> I am not sure whether this is related to this issue, but `expo publish` also seems to just create a bundle for iOS, not for android (when inspecting the json, there is just a ios URL). If it's a different issue, I will open a new one.

Each platform has its own entirely separate manifest; you can see the manifest for Android, which includes a link to the bundle created for Android, by appending `&platform=android` to the manifest URL or sending the `expo-platform: android` header. 🙂

**Comment by esamelson** [comment]
Hey @hannojg -- this is fixed in `expo-dev-client@0.4.7` which is out now. I would also recommend upgrading to `expo-updates@0.8.4`. Thanks again for the report!
